examples/source-map/webpack.config.js

```diff
@@ -1,27 +1,34 @@
 var path = require("path");
 
 module.exports = [
```

Comment:
You could write the config like that:

``` js
module.exports = (env = "development") => ({
  mode: env,
  devtool: env === "development" ? "cheap-eval-source-map" : "source-map",
  entry: "./example.coffee",
  module: {
    rules: [
      { test: /\.coffee$/, use: "coffee-loader" }
    ]
  }
});
```

`optimization` and `output` could be omitted. This should be a more realistic usage.

Reply:
yep, But I export a function config, and log `env` param, is always `undefined`

```js
module.exports = env => {
	return {
		mode: env,
		entry: "./example.coffee",
		devtool: env === "development" ? "cheap-eval-source-map" : "source-map",
		module: {
			rules: [{ test: /\.coffee$/, use: "coffee-loader" }]
		}
	};
};
```

when I set the default value is `development`, is always `development`, where did I write it wrong?

and I change `build.common.js` add `--env env mode`, it's be ok, but I found development result file `main.js` worng, it's not `cheap-eval-source-map` devtool

```js
async.series([
	callback => doCompileAndReplace("--mode production --env production", "production", callback),
	callback => doCompileAndReplace("--mode development --env development --devtool none", "development", callback),
	callback => doCompileAndReplace("--mode none --output-pathinfo", "", callback)
], () => {
	readme = tc.replaceBase(readme);
	fs.writeFile("README.md", readme, "utf-8", function() {});
});
```

Reply:
webpack cli `--devtool none` priority higher